tests/runtime/test_http_server.py

```diff
@@ -0,0 +1,255 @@
+                    "client",
+                    workdir,
+                    *tls_test_tools.generate_derived_key_cert_pair(ca_key=ca_key),
+                )
+                tls_config.client = KeyPair(cert=ca_certfile, key="")
+                # need to save the client cert and key in config_overrides so the mtls test below can access it
+                config_overrides["use_in_test"]["client_cert"] = client_certfile
+                config_overrides["use_in_test"]["client_key"] = client_keyfile
+
+            config_overrides["runtime"] = {"tls": tls_config}
+        port = http_server.RuntimeServerBase._find_port()
+        config_overrides.setdefault("runtime", {})["http"] = {
+            "timeout": 0.1,  # this is so the server is killed after 0.1 if no test is running
+            "port": port,
+            **http_config_overrides,
+        }
+
+        with temp_config(config_overrides, "merge"):
+            yield config_overrides
+
+
+## Insecure and TLS Tests #######################################################################
+
+
+def test_insecure_server():
+    with generate_tls_configs():
+        insecure_http_server = http_server.RuntimeHTTPServer()
+        with insecure_http_server.run_in_thread():
+            resp = requests.get(f"http://localhost:{insecure_http_server.port}/docs")
+            resp.raise_for_status()
+
+        insecure_http_server._stop_metering()
+
+
+def test_basic_tls_server():
+    with generate_tls_configs(
+        tls=True, mtls=False, http_config_overrides={}
+    ) as config_overrides:
+        http_server_with_tls = http_server.RuntimeHTTPServer(
+            tls_config_override=config_overrides["runtime"]["tls"]
+        )
+        with http_server_with_tls.run_in_thread():
+            resp = requests.get(
+                f"https://localhost:{http_server_with_tls.port}/docs",
+                verify=config_overrides["use_in_test"]["ca_cert"],
+            )
+            resp.raise_for_status()
+        http_server_with_tls._stop_metering()
+
+
+def test_mutual_tls_server():
+    with generate_tls_configs(
+        tls=True, mtls=True, http_config_overrides={}
+    ) as config_overrides:
+        http_server_with_mtls = http_server.RuntimeHTTPServer(
+            tls_config_override=config_overrides["runtime"]["tls"]
+        )
+        with http_server_with_mtls.run_in_thread():
+            resp = requests.get(
+                f"https://localhost:{http_server_with_mtls.port}/docs",
+                verify=config_overrides["use_in_test"]["ca_cert"],
+                cert=(
+                    config_overrides["use_in_test"]["client_cert"],
+                    config_overrides["use_in_test"]["client_key"],
+                ),
+            )
+            resp.raise_for_status()
+        http_server_with_mtls._stop_metering()
+
+
+# Third Party
+## Tests #######################################################################
+from fastapi.testclient import TestClient
```

Comment:
Let's move this import up to the top and update the comment bar to differentiate from the tests further up